service/src/main/java/bio/terra/landingzone/job/LandingZoneJobService.java

```diff
@@ -415,13 +425,19 @@ public void verifyUserAccessForDeleteJobResult(
         throw new JobNotFoundException(
             "The landing zone does not exist for job or the landing zone id is invalid.");
       }
+      var azureLandingZoneRequest =
+          inputParameters.get(
+              LandingZoneFlightMapKeys.LANDING_ZONE_CREATE_PARAMS, LandingZoneRequest.class);
 
-      // If the flight is completed successfully only check if the user is a valid SAM user
-      if (getJobStatus(flightState.getFlightStatus()).equals(JobReport.StatusEnum.SUCCEEDED)) {
```

Comment:
Doesn't this revert the bug? If the LZ is deleted, then the Sam check on L444 would fail, right?

Maybe need something like:
if (getJobStatus(flightState.getFlightStatus()).equals(JobReport.StatusEnum.SUCCEEDED)) {
  // check billing-profile
} else {
  // check landing-zone
}
```

Reply:
Let me clarify one thing. Once SAM resource is deleted it means that user lost permissions for this resource. Is that correct?

Reply:
I can see the point of using the SAM lz resource as much as possible, and only revert to the billing profile as a fallback. A few things consider then:

-  Is it safer to check the billing profile when in failed state - thinking of an edge case where the flight fails to persist the db state but the resource is already deleted?  If so, a more robust implementation can be:

```java
 if (status. equals(RUNNING)){
    //check LZ
   } else {
  //check billing profile
}
```


-Thinking of edge cases.... Do we need to worry about race conditions where flight status hasn't been updated and the SAM  lz resource does not exist - e.g. the client polls exactly when the deletion of the resource completes but before the status in the db changes?

Reply:
Yeah, once the Sam resource is deleted the user has lost permission to the resource -- e.g. `samService.checkAuthz` would throw `ForbiddenException`. The problem is that we still want the user to be able to check the status of the job, even if the Sam resource is gone.

The logic above seems okay -- I agree there could be race conditions though, if the Sam resource has been deleted but the flight status hasn't been updated yet.

Thinking out loud, another approach could be: 

```
// delete flight:
if job succeeded: 
  check bp 
else: 
  check lz

// create flight:
if job failed: 
  check bp
else: 
  check lz
```